staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -246,19 +244,11 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		// only match /apis/<group>/<version>
 		// only registered under /apis
 		if len(pathParts) == 3 {
-			if !r.hasSynced() {
-				responsewriters.ErrorNegotiated(serverStartingError(), Codecs, schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}, w, req)
-				return
-			}
 			r.versionDiscoveryHandler.ServeHTTP(w, req)
 			return
 		}
 		// only match /apis/<group>
 		if len(pathParts) == 2 {
-			if !r.hasSynced() {
-				responsewriters.ErrorNegotiated(serverStartingError(), Codecs, schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}, w, req)
-				return
-			}
```

Comment:
groupDiscoveryHandler would return what for unknown groups?

Reply:
it would return `503`, the `groupDiscoveryHandler` delegates to our custom handler. This scenario is covered by [a unit tests ](https://github.com/kubernetes/kubernetes/blob/86d845865a2a114eda94ecf7b491ad91d49e9911/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler_test.go#L237)

Reply:
ack